packages/google_maps_flutter/google_maps_flutter/example/integration_test/src/tiles_inspector.dart

```diff
@@ -204,6 +207,262 @@ void runTests() {
       },
     );
   }, skip: isWeb /* Tiles not supported on the web */);
+
+  /// Check that two lists of [WeightedLatLng] are more or less equal.
+  void expectHeatmapDataMoreOrLessEquals(
+    List<WeightedLatLng> data1,
+    List<WeightedLatLng> data2,
+  ) {
+    expect(data1.length, data2.length);
+    for (int i = 0; i < data1.length; i++) {
+      final WeightedLatLng wll1 = data1[i];
+      final WeightedLatLng wll2 = data2[i];
+      expect(wll1.weight, wll2.weight);
+      expect(wll1.point.latitude, moreOrLessEquals(wll2.point.latitude));
+      expect(wll1.point.longitude, moreOrLessEquals(wll2.point.longitude));
+    }
+  }
+
+  /// Check that two [HeatmapGradient]s are more or less equal.
+  void expectHeatmapGradientMoreOrLessEquals(
+    HeatmapGradient? gradient1,
+    HeatmapGradient? gradient2,
+  ) {
+    if (gradient1 == null || gradient2 == null) {
+      expect(gradient1, gradient2);
+      return;
+    }
+    expect(gradient2, isNotNull);
+
+    expect(gradient1.colors.length, gradient2.colors.length);
+    for (int i = 0; i < gradient1.colors.length; i++) {
+      final HeatmapGradientColor color1 = gradient1.colors[i];
+      final HeatmapGradientColor color2 = gradient2.colors[i];
+      expect(color1.color, color2.color);
+      expect(
+        color1.startPoint,
+        moreOrLessEquals(color2.startPoint, epsilon: floatTolerance),
+      );
+    }
+
+    expect(gradient1.colorMapSize, gradient2.colorMapSize);
+  }
+
+  void expectHeatmapEquals(Heatmap heatmap1, Heatmap heatmap2) {
+    expectHeatmapDataMoreOrLessEquals(heatmap1.data, heatmap2.data);
+    expectHeatmapGradientMoreOrLessEquals(heatmap1.gradient, heatmap2.gradient);
+    // Only Android supports `maxIntensity`
+    // so the platform value is undefined on others.
+    if (Platform.isAndroid) {
```

Comment:
nit/nonblocking 
Can you extract this comment and platform check into a method called canHandleMaxIntensity
```suggestion
    if(canHandleMaxIntensity()) {
```
Later 
```
    // Only Android supports `maxIntensity`
    // so the platform value is undefined on others.
    bool canHandleMaxIntensity(){
        return Platform.isAndroid;
    }
    // Only iOS supports `minimumZoomIntensity` and `maximumZoomIntensity`
    // so the platform value is undefined on others.
    bool canHandleMaximumZoomIntensity() {
        return Platform.isIOS;
    }
```
This makes it easier to reason about the code and if android or ios add the parameter later the modification is more contained.